gridscale/service-query/blocker.go

```diff
@@ -0,0 +1,140 @@
+package service_query
+
+import (
+	"fmt"
+	"github.com/gridscale/gsclient-go"
+	"github.com/hashicorp/terraform/helper/resource"
+	"time"
+)
+
+const (
+	delayFetchingStatus  = 500 * time.Millisecond
+	timeoutCheckDeletion = 1 * time.Minute
+)
+
+const (
+	provivisoningStatus = "in-provisioning"
+	activeStatus        = "active"
+)
+
+type gsService string
+
+const (
+	LoadbalancerService gsService = "loadbalancer"
+	IPService           gsService = "IP"
+	NetworkService      gsService = "network"
+	ServerService       gsService = "server"
+	SSHKeyService       gsService = "sshkey"
+	StorageService      gsService = "storage"
+	ISOImageService     gsService = "isoimage"
+)
+
+//BlockProvisoning blocks until the object's state is not in provisioning anymore
+func BlockProvisoning(client *gsclient.Client, service gsService, id string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		time.Sleep(delayFetchingStatus)
```

Comment:
I do not think `time.Sleep(delayFetchingStatus)`  is need. the retry will be triggered after this time `timeout` and forever. I think we need a way to stop the retry after awhile e.g, 10 minutes.

Reply:
I tried the code you gave me and it stops after the timeout.
`
return resource.Retry(10*time.Second, func() *resource.RetryError {
		time.Sleep(1 * time.Second)
		err := client.DeleteLoadBalancer(d.Id())

		return resource.RetryableError(fmt.Errorf("Error getting loadbalancer: %s", err))
	})
`

Reply:
I still keep `time.Sleep` as a rate limiter like `WaitForRequestCompletion` and `WaitForServerPowerStatus` in `gsclient-go`.